Simplify dataloaderutil

danamansana opened the pull request:
**What's this do?**

Two separate things, which could be separated if we want: 

1. Simplifies the `dataloaderutil`. Includes changes from `hold-request-with-redux-no-appdata`, but can be done independently

2. Call actions directly on the server side instead of passing around res.data. Personally I feel this is less hacky

**Did someone actually run this code to verify it works?**
PR author tested locally.

siberry commented on src/app/components/DataLoader/DataLoader.jsx:
```diff
@@ -21,4 +22,4 @@ DataLoader.contextTypes = {
 };
 
-export default DataLoader;
+export default connect()(DataLoader);
```
`connect` is not needed here

danamansana replied:
I think it is needed to make `dispatch` available in the props?

siberry replied:
Got it. It is possible to get `dispatch` by importing the store to `dataLoaderUtil`. But for some reason, that does not work for no-JS

siberry commented on src/app/stores/Store.js:
```diff
@@ -1,11 +1,12 @@
 }
 
-const store = configureStore(preloadedState);
+const store = configureStore(preloadedState || initialState);
```
How come you added `initialState` here?

danamansana replied:
This may be leftover from previous work. Will check if it is really needed and remove if not.

siberry commented on src/app/utils/dataLoaderUtil.js:
```diff
@@ -3,122 +3,61 @@ import {
-    return ajaxCall(apiRoute(matchData, route), successCb, errorCb);
-  }
-  return new Promise(resolve => resolve());
+  return ajaxCall(
```
This is now for server side as well, right?

danamansana replied:
Server side doesn't really make any API calls directly (and the API calls it makes indirectly already have their own error handling), so I don't think this is needed server side.

danamansana replied:
Realized what I wrote was a little cryptic. On the server side, `dataLoaderUtil` now never gets called. All that happens is that the express router recognizes the path (because of the configuration in `ApiRoutes`), passes the params to the corresponding function, and then passes the result directly into `successCb`, which puts them in the store.

siberry commented:
Also, could you remove dataLoaderUtil tests that do not make sense anymore? I could try writing new tests. But would be helpful, since that is a very long test file.

danamansana commented:
Good point about comments, will add some in. I think the tests will have to be rewritten so I can do that.

danamansana commented:
Rewrote the `dataLoaderUtil` tests. The tests for the `DataLoader` component are failing only because of the way the store is being passed, I think you may be better placed to deal with that because you are writing the redux tests for the other components (but let me know if you need me to take a look at it).

Also added some comments but not sure what will be most useful. Let me know if you think they should be changed.

siberry commented:
I approved, since I'd like to merge these changes into `switch-to-redux`. I'm not sure if there will be merge conflicts, but I can deal with those.
